src/librustc_mir/borrow_check/borrow_set.rs

```diff
@@ -254,15 +254,16 @@ impl<'a, 'gcx, 'tcx> Visitor<'tcx> for GatherBorrows<'a, 'gcx, 'tcx> {
         context: PlaceContext<'tcx>,
         location: Location,
     ) {
-        self.super_place(place, context, location);
+        let neo_place = self.tcx.as_new_place(place);
+        self.super_neoplace(&neo_place, context, location);
 
         // We found a use of some temporary TEMP...
-        if let Place::Local(temp) = place {
+        if let mir::PlaceBase::Local(temp) = &neo_place.base {
             // ... check whether we (earlier) saw a 2-phase borrow like
             //
             //     TMP = &mut place
-            match self.pending_activations.get(temp) {
-                Some(&borrow_index) => {
+            if neo_place.has_no_projection() {
```

Comment:
Ah well I guess this makes it equal to before, I take it back, but it still feels inelegant this way.